pybind/wallet3/wallet.cpp

```diff
@@ -0,0 +1,28 @@
+#include "../common.hpp"
+
+#include <wallet3/wallet.hpp>
+#include <wallet3/default_daemon_comms.hpp>
+#include <wallet3/keyring.hpp>
+#include <wallet3/config/config.hpp>
+#include <oxenmq/oxenmq.h>
+
+namespace wallet3
+{
+  void
+  Wallet_Init(py::module& mod)
+  {
+    py::class_<wallet::Wallet, std::shared_ptr<wallet::Wallet>>(mod, "Wallet")
+      .def(py::init([](std::string& wallet_name, wallet::Keyring& kr, wallet::Config& config) {
+        auto keyring = std::make_shared<wallet::Keyring>(std::move(kr));
```

Comment:
This doesn't seem right; we're nuking the caller's keyring out from under them.  Since Keyring is already using a shared_ptr holder, you can change the second argument here to a `std::shared_ptr<wallet::Keyring> keyring` and then just share ownership of the same keyring via the shared_ptr you get.